src/watcher/reconcilers/pods.go

```diff
@@ -271,7 +271,7 @@ func (p *PodWatcher) InitIntentsServerIndices(mgr ctrl.Manager) error {
 			}
 
 			for _, intent := range intents.GetCallsList() {
-				res = append(res, intent.Name)
+				res = append(res, intent.GetServerFullyQualifiedName(intent.GetServerFullyQualifiedName(intents.Namespace)))
```

Comment:
I guess the double call to `GetServerFullyQualifiedName` is a bug. 
Even if not I suggest a better practice of assigning to a local variable. Less cognitive load on the reader and shorter lines:

```suggestion
				fullName := intent.GetServerFullyQualifiedName(intents.Namespace)
				res = append(res, fullName)
```